python_scripts/database_builder_v2.py

```diff
@@ -513,43 +503,48 @@ def delete_database_data(db_path):
     wr.s3.delete_objects(db_path)
 
 
-def rebuild_database(db_name, rebuild_db, delete_underlying_data=True):
+def rebuild_database(
+        database_name: str, 
+        delete_data: bool = False, 
+        rename_db: Optional[str] = None):
     """
-    Using AWS wrangler to build database
-        Optionally drop
-    Paramaters
-    db_name (str): Name of database
-    rebuild_db (bool): True if database should be deleted
+    Using AWS wrangler to build database. 
+    Optionally drop data. 
+    Optionally rename database. 
+    
+    Args:
+        database_name (str): Name of current database
+        delete_data (bool): True if underlying data should be deleted
+        rename_db (str) New database name
     """
-
     # s3 client for handling exceptions
     s3 = boto3.client("s3")
 
```

Comment:
Does this function need some additional upfront logic i.e. if rename_db is None or rename_db == database_name then if database_name exists in glue catalog do nothing?

Reply:
Would it be easy to separate database_builder from this PR? I know that this function needs fixed, but don't want it to hold up automating the refresh of the schema.